pkg/expr/nodes.go

```diff
@@ -258,6 +261,24 @@ func (dn *DSNode) Execute(ctx context.Context, now time.Time, _ mathexp.Vars, s
 		return mathexp.Results{}, QueryError{RefID: dn.refID, Err: response.Error}
 	}
 
+	if !s.features.IsEnabled(featuremgmt.FlagDisableSSEDataplane) {
+		k, use, err := shouldUseDataplane(response.Frames)
+		if use {
+			if err != nil {
+				var vw *sdata.VersionWarning
+				if errors.As(err, &vw) {
+					logger.Warn("attempting to read mismatched version dataplane data", "error", err)
+				}
```

Comment:
I wonder if propagating logger to `shouldUseDataplane` and emitting a message in it would simplify the logic here. The reason is that you seem to make the the special case for error in two places just for the sake of logging.